File: C#/04 - Log Levels/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int separatorIndex = logLine.IndexOf(": ");
        return separatorIndex == -1 ? logLine : logLine.Substring(separatorIndex + 2).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int start = logLine.IndexOf("[") + 1;
        int end = logLine.IndexOf("]");
        return logLine.Substring(start, end - start).ToLower();
    }

    public static string Reformat(string logLine)
    {
        string message = Message(logLine);
        string level = LogLevel(logLine);
        return $"{message} ({level})";
    }
}

File: Java/02 - Log Levels/LogLevels.java
public class LogLevels {
    public static String message(String logLine) {
        int index = logLine.indexOf("]: ");
        String result = logLine.substring(index + 3);
        return result.trim();
    }

    public static String logLevel(String logLine) {
        int start = logLine.indexOf("[") + 1;
        int end = logLine.indexOf("]");
        String level = logLine.substring(start, end).toLowerCase();
        return level;
    }

    public static String reformat(String logLine) {
        String a = message(logLine);
        String b = logLevel(logLine);

        return a + " (" + b + ")";
    }
}